cirq/ops/named_qubit.py

```diff
@@ -33,29 +37,112 @@ def __init__(self, name: str) -> None:
+    """
+
+    def __init__(self, name: str, dimension: int) -> None:
+        super().__init__(name)
+        self._dimension = dimension
+        self.validate_dimension(dimension)
+
     @property
     def dimension(self) -> int:
-        return 2
+        return self._dimension
 
-    def __str__(self) -> str:
+    @property
+    def name(self) -> str:
         return self._name
 
+    def _with_name(self, name: str) -> 'NamedQid':
+        return NamedQid(name, dimension=self.dimension)
+
+    def __repr__(self) -> str:
+        return f'cirq.NamedQid({self.name!r}, ' f'dimension={self.dimension})'
+
+    def __str__(self) -> str:
+        return f'{self.name} (d={self.dimension})'
+
+    @staticmethod
+    def range(*args, prefix: str, dimension: int) -> List['NamedQid']:
+        """Returns a range of NamedQids.
+
+        The range returned starts with the prefix, and followed by a qid for
+        each number in the range, e.g.:
+
+        NamedQid.range(3, prefix='a') -> ['a1', 'a2', 'a3']
+        NamedQid.range(2, 4, prefix='a') -> ['a2', 'a3']
+
+        Args:
+            *args: Args to be passed to Python's standard range function.
+            prefix: A prefix for constructed NamedQids.
+
+        Returns:
+            A list of NamedQids.
+            """
+        return [
+            NamedQid(prefix + str(i), dimension=dimension) for i in range(*args)
+        ]
+
+    def _json_dict_(self) -> Dict[str, Any]:
+        return protocols.obj_to_dict_helper(self, ['name', 'dimension'])
+
+
+class NamedQubit(_BaseNamedQid):
+    """A qubit identified by name.
+
+    By default, NamedQubit has a lexicographic order. However, numbers within
+    the name are handled correctly. So, for example, if you print a circuit
+    containing `cirq.NamedQubit('qubit22')` and `cirq.NamedQubit('qubit3')`, the
+    wire for 'qubit3' will correctly come before 'qubit22'.
+    """
+
     @property
-    def name(self) -> str:
+    def dimension(self) -> int:
+        return 2
+
+    def _cmp_tuple(self):
+        cls = NamedQid if type(self) is NamedQubit else type(self)
+        # Must be same as Qid._cmp_tuple but with cls in place of type(self).
+        return (cls.__name__, repr(cls), self._comparison_key(), self.dimension)
+
+    def _with_name(self, name: str) -> 'NamedQubit':
+        return NamedQubit(name)
+
+    def __str__(self) -> str:
         return self._name
 
     def __repr__(self) -> str:
         return f'cirq.NamedQubit({self._name!r})'
 
     @staticmethod
-    def range(*args, prefix: str):
+    def range(*args, prefix: str) -> List['NamedQubit']:
         """Returns a range of NamedQubits.
 
         The range returned starts with the prefix, and followed by a qubit for
         each number in the range, e.g.:
 
-        NamedQubit.range(3, prefix="a") -> ["a1", "a2", "a3]
-        NamedQubit.range(2, 4, prefix="a") -> ["a2", "a3]
+        NamedQubit.range(3, prefix='a') -> ['a1', 'a2', 'a3']
```

Comment:
keep double quotes

Reply:
Good question. I might have jumped the gun here without a solid foundation, as I thought it was an unnecessary find/replace from double to single. Let me find guidance on this.

Reply:
The only guidance in https://google.github.io/styleguide/pyguide.html is to stay consistent within a file: 

> Be consistent with your choice of string quote character within a file. Pick ' or " and stick with it. It is okay to use the other quote character on a string to avoid the need to \\ escape within the string.